catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -28,17 +28,27 @@
 import android.app.PendingIntent;
 import android.content.Context;
 import android.content.Intent;
+import android.content.SharedPreferences;
 import android.media.RingtoneManager;
 import android.net.Uri;
+import android.preference.PreferenceManager;
+import android.support.annotation.VisibleForTesting;
 
+import com.google.firebase.messaging.FirebaseMessaging;
+
+import org.catrobat.catroid.BuildConfig;
 import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.SettingsActivity;
 import org.catrobat.catroid.utils.StatusBarNotificationManager;
 
 public class CloudMessaging {
 
+	public static final String DEFAULT_TOPIC = "DEFAULT_TOPIC";
+
 	private final Context context;
 	private final NotificationBuilderProvider notificationBuilderProvider;
 	private final NotificationManager notificationManager;
+	private static boolean isCloudNotificationsEnabled = BuildConfig.FEATURE_CLOUD_MESSAGING_ENABLED;
```

Comment:
We implemented the CrashReporter the same way. I haven't found a better solution yet. Can you please also do some research here?